numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -3,14 +3,48 @@
 import os
 from collections import namedtuple
 import platform
-
+import site
+from pathlib import Path
 from numba.core.config import IS_WIN32
-from numba.misc.findlib import find_lib, find_file
+from numba.misc.findlib import find_lib
 from numba import config
-
+import ctypes
 
 _env_path_tuple = namedtuple("_env_path_tuple", ["by", "info"])
 
+SEARCH_PRIORITY = [
+    "Conda environment",
+    "Conda environment (NVIDIA package)",
```

Comment:
Leaving a note here, no action needed for this PR. 

I understand this is just a refactoring of existing code, but still having this distinction (conda-forge/nvidia) is a bit nerve wrecking, especially after CUDA 12.0 where both channels started unifying the layouts (different from `cudatoolkit` from 11.x and before) and eventually became interchangeable around 12.5. cc @jakirkham for comments.

Reply:
This distinction is here because there's a distinction in the packages though - we can't unify it here because it's not unified in the wider landscape of CUDA toolkit conda packaging.

Reply:
I assume you're referring to the layout differences between CUDA 11/12?